storage/innobase/include/dict0mem.h

```diff
@@ -2138,6 +2141,12 @@ struct dict_table_t {
 	It is protected by lock_sys.mutex. */
 	ulint					n_rec_locks;
 
+	bool                           alter_persistent_count;
+	/** Mutex for committed_count_inited accesses.*/
+	ib_mutex_t				       committed_count_mutex;
+	bool                           committed_count_inited;
+	Atomic_counter<ib_uint64_t>    committed_count;
+
```

Comment:
I do not like storing transient information in a persistent object. Do we really need `dict_table_t::alter_persistent_count`?

Can we protect `committed_count_inited` with some existing synchronization device, such as `dict_index_t::lock` or MDL?

Side note: I realize that the concurrency protection rules around the `dict_index_t` bit-fields are not documented clearly, and I see that `unsigned online_status:2` should probably reside in its own data field, because some changes are protected by `dict_index_t::lock` while most everything else is protected by the `dict_sys` latches.

Reply:
> I do not like storing transient information in a persistent object. Do we really need dict_table_t::alter_persistent_count?

Agreed, that was one of my questions. `dict_table_t::alter_persistent_count` is solely used to prevent `innobase_update_persistent_count(...)` from redundantly executing for an `ALTER TABLE {table} ENABLE|DISABLE PERSISTENT_COUNT` statement. Is there a clean workaround you can think of for this?

Reply:
I added a `uncommitted_count != 0` check before calling `innobase_update_persistent_count()`, which resolves the above concern.